p8e-shared/src/main/kotlin/io/provenance/p8e/shared/service/AffiliateService.kt

```diff
@@ -53,17 +59,54 @@ class AffiliateService(
         const val AFFILIATE_ENCRYPTION_KEY_PAIR = "affiliate_encryption_key_pair"
         const val AFFILIATE_BECH32_LOOKUP = "affiliate_bech32_lookup"
         const val PUBLIC_KEY_TO_ADDRESS = "public_key_to_address"
+        const val AFFILIATE_SIGNING_KID = "affiliate_signing_key_id"
+        const val AFFILIATE_ENCRYPTION_PUBLIC_KEY = "affiliate_encryption_public_key"
+    }
+
+    /**
+     *  Return the signer to be used
+     *
+     *  [publicKey] publicKey used to validate
+     *  [signer] The specific signer used to sign contracts
+     */
+    fun getSigner(publicKey: PublicKey): SignerImpl {
+        val affiliateRecord = get(publicKey)
+        return when(affiliateRecord?.keyType) {
+            SMARTKEY -> signerFactory.getSigner(SignerFactoryParam.SmartKeyParam(affiliateRecord.signingKeyUuid.toString()))
+            DATABASE -> signerFactory.getSigner(SignerFactoryParam.PenParam(KeyPair(affiliateRecord.publicKey.value.toJavaPublicKey(), affiliateRecord.privateKey!!.toJavaPrivateKey())))
+            else -> throw UnsupportedOperationException("Unsupported signer type found ${affiliateRecord?.keyType?.name}.")
```

Comment:
You should be able to remove this case. Looks like it's only needed when affiliateRecord is null but in that case we can return an affiliate not found or something before the `when` expression.